pydis_site/apps/content/resources/guides/python-guides/mutability.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Mutability and Immutability in Python
+description: "Mutable and immutable data types: What they are and how they work."
+---
+
+Consider this example:
+```python
+>>> s = "hello"
+>>> s.upper()
+'HELLO'
+>>> s
+'hello'
+```
+This might break your expectations.
+After all, you've called the `upper()` method on `s`, so why didn't it change? That's because strings are _immutable_: you can't change them in-place, only create new ones.
+In this example, `.upper()` just cannot change the string stored in `s`.
+
+How do you make `s` store `'HELLO'` instead of `'hello'` then? That's possible.
+Even though you can't change the original string, you can create a new one, which is like the old one, but with all letters in upper case.
+
+In other words, `s.upper()` doesn't change an existing string.
+It just returns a new one.
+```python
+>>> s = 'hello'
+>>> s = s.upper()
+>>> s
+'HELLO'
+```
+
+Let's examine what's going on here.
+At first, the variable `s` refers to some object, the string `'hello'`.
+
+![s refers to the string "hello"](/static/images/content/mutability/s_refers_hello.webp)
+
+When you call `s.upper()`, a new string, which contains the characters `'HELLO'`, gets created.
+![s.upper creates "HELLO"](/static/images/content/mutability/s_upper_creates_HELLO.webp)
```

Comment:
The images in this page are misaligned.
![Screenshot_2021-04-17 Python Discord - Mutability and Immutability in Python](https://user-images.githubusercontent.com/1515135/115124443-1fe5a380-9f77-11eb-8c00-e36add6546a1.png)

Reply:
Addressed in a782abd.